### PR TITLE
[MRG+1] FIX markerfacecolor / mfc not in rcparams

### PR DESCRIPTION
**Pull request by kalagau**
PR for issue #3293 and #8109  as well as suggested by @anntzer

### CODE REVIEW

**Comment by tacaswell** [comment]
The test failures look real, can you reproduce them locally?

https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/tests/baseline_images/test_artist/default_edges.png that is the test that is changed.

**Comment by kalagau** [comment]
default_edges
default_edges-expected
I can't seem to notice the difference?
This is the diff.
default_edges-failed-diff

I ran this on 3.5 and 2.7

**Comment by anntzer** [comment]
There's also markeredgecolor (#8109).  Could you also take care of it?

**Comment by kalagau** [comment]
Sure, I am just trying to figure out the reason behind the tests failing before making anymore changes

**Comment by kalagau** [comment]
@tacaswell the fixes seem to pass both travis and appveyor now

**Comment by tacaswell** [comment]
:+1: 

Can you add a note in https://github.com/matplotlib/matplotlib/tree/master/doc/users/whats_new and add these values to https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/mpl-data/stylelib/classic.mplstyle ?

**Comment by kalagau** [comment]
Done!

**Comment by kalagau** [comment]
I wonder why sometimes there are 4 checks instead of 5?

**Comment by tacaswell** [comment]
Sorry, one more thing, could you add a test?  Something like

```python
def test_mfc_rcparams():

    rcParams['line.markerfacecolor'] = 'red'
    ln = Line2D([1, 2], [1, 2])
    assert ln.get_markerfacecolor() == matplotlib.colors.to_rgba('red')

```

and so-on for the two rcparams added here.

I suggest putting this test in `test_lines` or `test_rcparams`.

File: lib/matplotlib/tests/test_rcparams.py [hunk 1]
```diff
@@ -210,6 +210,19 @@ def test_legend_colors(color_type, param_dict, target):
         leg = ax.legend()
         assert getattr(leg.legendPatch, get_func)() == target
 
+@cleanup
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_rcparams.py]: Don't need cleanup anymore on master :)

**Comment by tacaswell** [comment]
Assuming they pass, yes :+1:

**Comment by kalagau** [comment]
Great 👍

**Comment by kalagau** [comment]
All tests passed! 👍

**Comment by kalagau** [comment]
Is anything else required here? Or is this good to go as well?

File: doc/users/whats_new/new_rcparams_added.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+New rcParams added
+----------------------------------------------------------------------
```
Comment (QuLogic) [inline doc/users/whats_new/new_rcparams_added.rst]: Trim dashes to text length.

File: doc/users/whats_new/new_rcparams_added.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+* 'lines.markerfacecolor' allows users to set the default markerfacecolor for line plots
+* 'lines.markeredgecolor' allows users to set the default markeredgecolor for line plots
+The Default color is set to none, and these rcParams are only used if the line does not have any value specifed for these params.
```
Comment (QuLogic) [inline doc/users/whats_new/new_rcparams_added.rst]: Please wrap at 80 here as well. This line looks like it should be part of the previous bullet point, so add a new line above it.

File: lib/matplotlib/lines.py [hunk 4]
```diff
@@ -339,6 +339,10 @@ def __init__(self, xdata, ydata,
             linestyle = rcParams['lines.linestyle']
         if marker is None:
             marker = rcParams['lines.marker']
+        if markerfacecolor is None and not rcParams['lines.markerfacecolor'] == 'none':
```
Comment (QuLogic) [inline lib/matplotlib/lines.py]: `rcParams['lines.markerfacecolor'] != 'none'`

File: lib/matplotlib/lines.py [hunk 5]
```diff
@@ -339,6 +339,10 @@ def __init__(self, xdata, ydata,
...
+            markerfacecolor = rcParams['lines.markerfacecolor']
+        if markeredgecolor is None and not rcParams['lines.markeredgecolor'] == 'none':
```
Comment (QuLogic) [inline lib/matplotlib/lines.py]: Same as above.

**Comment by kalagau** [comment]
I used no color as a default to be able to tell if the user has specified a value for the rcParam value.

**Comment by kalagau** [comment]
I believe 3.4 is the buggy run, could someone please confirm.

**Comment by kalagau** [comment]
@QuLogic your suggestions were implemented. I think this should be good now unless anyone else has some suggestions?

**Comment by QuLogic** [comment]
It seems okay, except I'm still confused about the semantics, though maybe I'm just mistaken about how rcParam defaults work. @tacaswell?

**Comment by kalagau** [comment]
By initially setting the rcParam to 'none', what essentially ends up happening is that the line gets drawn the same way when the user does not specify the color.

I think what you were perhaps implying before is that 'none' is a color option where no color is filled and by selecting that as a default I perhaps take away the user option to use that?

**Comment by tacaswell** [comment]
> I think what you were perhaps implying before is that 'none' is a color option where no color is filled and by selecting that as a default I perhaps take away the user option to use that?

Yes 'none' is a valid color meaning "no color" (which is different than None).  The default values should be the current default value.

**Comment by tacaswell** [comment]
@kalagau Sorry this sat so long!  Are you still interested in working on this?

**Comment by kalagau** [comment]
Hey Sorry for the delay @tacaswell.

Ill take a look at this shortly

**Comment by kalagau** [comment]
@tacaswell I have updated the PR.  Assuming all tests pass, I feel that this should be fine now?

**Comment by kalagau** [comment]
I am not sure why Travis CI failed?

**Comment by jklymak** [comment]
We had some test failures with OS X. The CI is clean.

I'm not merging yet because @phobson 's review is quite stale.  It'd be nice to have a more recent review, and marking as such.  @tacaswell is quite busy, so may dismiss his review if there is other support for this.

**Comment by kalagau** [comment]
@jklymak is there anything else we are waiting for?

**Comment by jklymak** [comment]
Thanks a lot for your patience and PR @kalagau!

**Comment by crstnbr** [comment]
Which matplotlib version contains/will contain this?

**Comment by jklymak** [comment]
3.0

**Pull request merged**